frontend/src/stores/upload.ts

```diff
@@ -47,6 +47,41 @@ export const useUploadStore = defineStore("upload", {
       const sum = state.progress.reduce((acc, val) => +acc + +val) as number;
       return Math.ceil((sum / totalSize) * 100);
     },
+    getProgressDecimal: (state) => {
+      if (state.progress.length === 0) {
+        return 0;
+      }
+
+      const totalSize = state.sizes.reduce((a, b) => a + b, 0);
+
+      // TODO: this looks ugly but it works with ts now
+      const sum = state.progress.reduce((acc, val) => +acc + +val) as number;
+      return ((sum / totalSize) * 100).toFixed(2);
+    },
+    getTotalProgressBytes: (state) => {
+     if (state.progress.length === 0 || state.sizes.length === 0) {
+        return '0 Bytes';
+      }
+    
+      const totalSize = state.sizes.reduce((a, b) => a + b, 0);
+
+      // TODO: this looks ugly but it works with ts now
+      const sum = state.progress.reduce((acc, val) => +acc + +val) as number;
+      const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
+      const i = Math.floor(Math.log(sum) / Math.log(1024));
+      return parseFloat((sum / Math.pow(1024, i)).toFixed(2)) + ' ' + sizes[i];
+      
+    },
+    getTotalSize: (state) => {
+      if (state.sizes.length === 0) {
+        return '0 Bytes';
+      }
+    
+      const totalSize = state.sizes.reduce((a, b) => a + b, 0);
+      const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
+      const i = Math.floor(Math.log(totalSize) / Math.log(1024));
+      return parseFloat((totalSize / Math.pow(1024, i)).toFixed(2)) + ' ' + sizes[i];
```

Comment:
This code segment is duplicated and can be extracted into a common function

Reply:
Went and implemented this! https://github.com/filebrowser/filebrowser/pull/3111/commits/d29dee3a5e1eb83fb139ffef774e9bb84a6af0ff